membership-attribute-service/app/models/Attributes.scala

```diff
@@ -19,14 +17,36 @@ object ContentAccess {
   implicit val jsWrite = Json.writes[ContentAccess]
 }
 
+case class CardDetails(last4: String, expirationMonth: Int, expirationYear: Int, forProduct: String) {
+  val asLocalDate = new LocalDate(expirationYear, expirationMonth, 1).plusMonths(1).minusDays(1)
```

Comment:
Oh! My preference runs the other way! I'd probably go with `val` by default - seeing as the value never changes no matter how many times you call it, so it _is_ constant and doesn't _need_ to be a `def`. If calculating the value was expensive (not the case here) I might go for a `lazy val`.